### PR TITLE
ENH: Remove TestDriver duplicate macro for Nonunit tests

### PR DESCRIPTION
**Pull request by tbirdso**
Draft PR removing `CreateTestDriver_SupportBuildInIOFactories` as a duplicate of `CreateTestDriver` as both register default factories via test kernel initialization. Addresses #3004.

Tests are passing locally on Windows, will examine CI to verify that tests pass across platforms. I haven't been able to find a particular reason for the duplicate macro in modern code, current hypothesis is that `CreateTestDriver_SupportBuildInIOFactories` was created for legacy reasons that are now outdated. Would be interested in the perspective of others with more insight into the history of `Nonunit` tests.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Modules/Core/TestKernel/include/itkTestDriverIncludeBuiltInIOFactories.h [hunk 1]
```diff
@@ -1,83 +0,0 @@
...
-
-#ifdef __EMSCRIPTEN__
-#  include <emscripten.h>
-#endif
-
-void
-ProcessArgumentsAndRegisterBuiltInFactories(int * ac, ArgumentStringType * av)
-{
-  // ImageIO
-  itk::ObjectFactoryBase::RegisterFactory(itk::BioRadImageIOFactory::New());
```
Comment (dzenanz) [inline Modules/Core/TestKernel/include/itkTestDriverIncludeBuiltInIOFactories.h:58]: I think the point of this was to test all the built-in IOs. I guess that its existence is now moot.

**Review by dzenanz — APPROVED** [review]
Hopefully no further changes are needed.

**Review by thewtex — APPROVED** [review]

File: Modules/Core/TestKernel/include/itkTestDriverIncludeBuiltInIOFactories.h (hunk 1 above)
Comment (tbirdso) [reply Modules/Core/TestKernel/include/itkTestDriverIncludeBuiltInIOFactories.h:58]: Sounds good. Given that tests pass after removing `itkTestDriverIncludeBuiltInIOFactories.h`, we can rely on `itkTestDriverIncludeRequiredFactories.cxx` for a factory testing.

**Comment by tbirdso** [comment]
Marked as ready after confirming tests have passed. @thewtex @dzenanz could this be merged when convenient? Thank you!

**Pull request merged**